app/app/route.js

```diff
@@ -15,8 +15,12 @@ export default Ember.Route.extend({
     return tokens.join(' - ');
   },
 
+  setupController: function(controller, model){
+    controller.set('model', model);
+    controller.set('stack', model.get('stack.content'));
```

Comment:
@sandersonet Yeah, sorry I didn't describe that better before. Because app.stack is an async relationship (the ember-data relationship is [defined here](https://github.com/aptible/diesel.aptible.com/blob/master/app/models/app.js#L19)), app.get('stack') is *always* going to return a promise proxy (whether the stack is loaded or not). This usually works fine in templates because the it will proxy all `get`s for properties to the content. But aptible-ability is checking for a `permitsRole` method directly on the promise proxy, which isn't there.

Usually it's an anti-pattern to access the content directly, because if the async relationship isn't loaded `app.get('stack.content')` will equal `undefined`.

Because of the `afterModel` call in this route, we can be certain that the app's stack is loaded, so `app.get('stack.content')` returns the actual instance of the `Stack` model rather than undefined.

There are 2 ways to get around accessing content directly that I can think of:
  * make aptible-ability "promise-aware", having it absorb that async and defer checking the ability until everything passed to it is loaded. This is a little complicated but if it makes ability-checking smoother it may be worthwhile.
  * stash the stack as a property on the route in afterModel:
    ```
      afterModel(model){
         return model.get('stack').then( (stack) => {
            this.stack = stack;
          });
          // in setupController, you can now do "controller.set('stack', this.stack);"
      }
   ```
   This isn't ideal because you are adding statefulness to the router.

cc @mixonic Any other ideas for best practices here?